src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala

```diff
@@ -0,0 +1,174 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package com.bwsw.kafka.reader
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import com.bwsw.kafka.reader.entities.{TopicInfo, TopicInfoList}
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.scalatest.{Outcome, fixture}
+
+import scala.util.{Failure, Success, Try}
+
+
+class KafkaReaderIntegrationTest extends fixture.FlatSpec {
+  var topicNumber = 0
+
+  case class FixtureParam(producer: Producer[String, String], kafkaEndpoints: String)
+
+  def withFixture(test: OneArgTest): Outcome = {
+    val kafkaEndpoints = ApplicationConfig.getRequiredString("app.kafka.endpoints")
+
+    val producer = new Producer[String, String](kafkaEndpoints)
+
+    val theFixture = FixtureParam(producer, kafkaEndpoints)
+
+    Try {
+      withFixture(test.toNoArgTest(theFixture))
+    } match {
+      case Success(x) =>
+        Try(producer.close())
+        x
+      case Failure(e: Throwable) =>
+        producer.close()
+        throw e
+    }
+  }
+
+  "MockEventHandler" should "handle all events from single Kafka topic and then handle nil events" in { fixture =>
+    val groupId = "group1"
+    val topic = getNextTopic
+    val topicInfoList = TopicInfoList(List(TopicInfo(topic)))
+    val countOfTestData = 10
+    val expectedTestDataList = createListWithTestData(topic, countOfTestData)
+    val producerRecords = expectedTestDataList.map {
+      case (topicData, value) => new ProducerRecord[String, String](topicData, 0, "key", value)
+    }
+
+    fixture.producer.send(producerRecords)
+
+    testForNonEmptyTopics(
+      fixture.kafkaEndpoints,
+      groupId,
+      topicInfoList,
+      countOfTestData,
+      expectedTestDataList.map(_._2).toSet
+    )
+
+    testForEmptyTopics(fixture.kafkaEndpoints, groupId, topicInfoList)
+  }
+
+  "MockEventHandler" should "handle all events from multiple Kafka topics and then handle nil events" in { fixture =>
```

Comment:
it should "process all events from multiple Kafka topics, save checkpoint data (indicating what events have been processed), run from a scratch and not receive any message"

Reply:
ok